core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -652,13 +678,27 @@ public static PropertyKey fromString(String keyStr) {
    * Constructs a configuration property.
    *
    * @param property String of this property
+   * @param defaultValue Default value of this property in compile time if not null
    */
-  PropertyKey(String property) {
-    mString = property;
+  PropertyKey(String property, Object defaultValue) {
+    mName = property;
+    if (defaultValue == null) {
+      mDefaultValue = null;
```

Comment:
default value of `null` is expected cases here.
some properties are not determined in compile time (e.g., master address), or platform dependent